EventBus.Kafka/KafkaEventSubscriber.cs

```diff
@@ -0,0 +1,85 @@
+﻿
+using System;
+using System.Threading.Tasks;
+using Finaps.EventBus.Kafka.Configuration;
+using Finaps.EventBus.Core;
+using Finaps.EventBus.Core.Abstractions;
+using Finaps.EventBus.Core.Utilities;
+using Microsoft.Extensions.Logging;
+using Confluent.Kafka;
+
+namespace Finaps.EventBus.Kafka
+{
+
+  internal class KafkaEventSubscriber : IEventSubscriber
+  {
+    private bool _disposed;
+    private readonly ILogger<KafkaEventSubscriber> _logger;
+    public event AsyncEventHandler<IntegrationEventReceivedArgs>? OnEventReceived;
+
+    private KafkaOptions _options;
+    private IConsumer<Ignore, string> _consumer;
+
+    internal KafkaEventSubscriber(
+      ILogger<KafkaEventSubscriber> logger,
+      KafkaOptions options)
+    {
+      _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+      _options = options;
+      _consumer = new ConsumerBuilder<Ignore, string>(
+        new ConsumerConfig
+        {
+          GroupId = options.GroupId,
+          BootstrapServers = options.Brokers,
+          AutoOffsetReset = AutoOffsetReset.Earliest,
+        }
+      ).Build();
+    }
+
+    public Task InitializeAsync()
+    {
+      return Task.CompletedTask;
+    }
+
+    public Task SubscribeAsync(string eventName)
+    {
+      _consumer.Subscribe(_options.TopicName);
+      return Task.CompletedTask;
+    }
+
+    public Task StartConsumingAsync()
+    {
+      Task.Run(async () => {
+        try
+        {
+          while (true)
+          {
+              var consumer = _consumer.Consume();
```

Comment:
Logic is missing here to route the consumed message. 
Somehow per the consumed message, we need to route them to the correct event. The other implementations do this by raising an event with the `EventName` & `Message` you can see this data structure within the core `IntegrationEventReceivedArgs`

However, per the documentation of the Kafka consumer the consumed message follows this structure: [consume result](https://github.com/confluentinc/confluent-kafka-dotnet/blob/master/src/Confluent.Kafka/ConsumeResult.cs)

The only real option I see for routing is doing something via headers? [Kafka Header](https://github.com/confluentinc/confluent-kafka-dotnet/blob/master/src/Confluent.Kafka/Header.cs)

Only downside is this approach isn't guarantied to work in all implementations.